### PR TITLE
Release snapshot workflow refactoring

### PR DESCRIPTION
**Pull request by SevaZhukov**
Failure release-snapshot job confuses the team when it runs on branches and PRs without the `publish-snapshot` label.

I propose a way like for the mobile-metrics-benchmarks job. If you want to publish a snapshot just approve the `release-snapshot-start` job, and it will start the `release-snapshot-1` job (with postfix 1 because we use this job for the main and release branches too). The `release-snapshot-2` job will be started without approvals for the main and release branches.

image

@RingerJK @Zayankovsky

### CODE REVIEW

**Review by dzinad — APPROVED** [review]

**Review by VysotskiVadim — APPROVED** [review]

**Comment by SevaZhukov** [comment]
Also, I've added the GitHub action which runs the `release-snapshot` job on every commit if you set the label `publish-snapshot`

**Review by RingerJK — APPROVED** [review]
🚀

File: .circleci/config.yml [hunk 1]
```diff
@@ -32,7 +32,22 @@ workflows:
...
+          type: approval
+          filters:
+            branches:
+              ignore:
+                - main
+                - /release-v.*/
+      - release-snapshot:
+          requires:
+            - release-snapshot-start
+      - release-snapshot:
```
Comment (dzinad) [inline .circleci/config.yml:45]: Why do you have 2 jobs with the same name?

Comment (SevaZhukov) [reply .circleci/config.yml:45]: This is one job. We run it by:
- main and release branches
- approval in circle ci
- set PR label

**Pull request merged**